src/js/keyboardInput.js

```diff
@@ -206,14 +216,18 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
                 }
             }
             if (ch !== "") {
-                input.val(ch);
-                // programmatic change of the input value does not
-                // fire a change event, so we trigger it explicitly
-                input.triggerHandler("change");
+                that.events.keypress.fire(ch);
             }
         });
     };
 
+    gpii.firstDiscovery.keyboardInput.setElementValueAndTriggerChange = function (elem, value) {
```

Comment:
We should take the opportunity to elevate this into a core utility within FluidDocument.js called, say, fluid.changeElementValue with its own tests. It appears in numerous places, for example in TestingTests.js utility "fluid.tests.changeField", TextfieldSliderTests.js manually, PanelsTests.js and many others.